doc/examples/transform/plot_transform_types.py

```diff
@@ -0,0 +1,142 @@
+shown here, by increasing order of complexity. While we focus here on the
+mathematical properties of transformations, tutorial :ref:`sphx_glr_auto_examples_transform_plot_geometric.py`
+explains how to use such transformations for various tasks such as image warping or parameter estimation.
+"""
+import math
+import numpy as np
+import matplotlib.pyplot as plt
+
+from skimage import data
+from skimage import transform as tf
+from skimage import img_as_float
+
+######################################################################
+# Euclidean (rigid) transformation
+# =================================
+#
+# A `Euclidean transformation <https://en.wikipedia.org/wiki/Rigid_transformation>`_,
+# also called rigid transformation, preserves the Euclidean distance between
+# pairs of points. It can be described as a rotation about the origin
+# followed by a translation.
+
+tform = tf.EuclideanTransform(
+    rotation=np.pi / 12.,
+    translation = (100, -20)
+    )
+print(tform.params)
+
+######################################################################
+# Now let's apply this transformation to an image. Because we are trying
+# to reconstruct the *image* after transformation, it is not useful to see
+# where a *coordinate* from the input image ends up in the output, which is
+# what the transform gives us. Instead, for every pixel (coordinate) in the
+# output image, we want to figure out where in the input image it comes from.
+# Therefore, we need to use the inverse of ``tform``, rather than ``tform``
+# directly.
+
+img = img_as_float(data.chelsea())
+tf_img = tf.warp(img, tform.inverse)
+_ = plt.imshow(tf_img)
+plt.title('Euclidean transformation')
+
+
+######################################################################
+# For a rotation around the center of the image, one can 
+# compose a translation to change the origin, a rotation, and finally
+# the inverse of the first translation.
+# in degrees, while it is given in radians for the different kinds of
+# affine transformations):
+
+rotation = tf.EuclideanTransform(rotation=np.pi/3)
+shift = tf.EuclideanTransform(translation=-np.array(img.shape[:2]) / 2)
+# Compose transforms by multiplying their matrices
+matrix = np.linalg.inv(shift.params) @ rotation.params @ shift.params
+tform = tf.EuclideanTransform(matrix)
+tf_img = tf.warp(img, tform.inverse)
+plt.figure()
+_ = plt.imshow(tf_img)
+
+######################################################################
+# Similarity transformation
+# =================================
+#
+# A `similarity transformation <https://en.wikipedia.org/wiki/Similarity_(geometry)>`_
+# preserves the shape of objects. It combines scaling, translation and rotation.
+
+tform = tf.SimilarityTransform(
+    scale=0.5,
+    rotation=np.pi/12,
+    translation=(100, 50))
+print(tform.params)
+tf_img = tf.warp(img, tform.inverse)
+plt.figure()
+plt.title('Similarity transformation')
+_ = plt.imshow(tf_img)
```

Comment:
Ah, sorry that I didn't catch the plotting before. Can we use the modern Matplotlib OO interface, rather than the state machine?

```python
fig, ax = plt.subplots()
ax.imshow(tf_img)
ax.set_title('Similarity transformation')
plt.show()
```

Reply:
(comment goes for the other sections as well, obviously)